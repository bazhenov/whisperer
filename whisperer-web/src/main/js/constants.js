export const SSE_RECONNECT_TIMEOUT = 1000;
export const LS_CONNECTION_PARAMS_KEY = 'connectionParams';
export const MAX_MESSAGES_COUNT = 50000;
export const MAX_MESSAGES_TO_DISPLAY = 1000;