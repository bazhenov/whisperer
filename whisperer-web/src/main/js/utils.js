import { Set, List, Map } from 'immutable';

export const getGroupShortName = (group) => {
	const parts = (group || '').split('.');
	return parts.reduce((name, part, i) => {
		if (part.length === 0) return name;
		return name + (name.length === 0 ? '' : '.') + (i === (parts.length - 1) ? part : part[0])
	}, '');
};

const isEmptyOrContains = (col, value) => {
	return col.isEmpty() || col.contains(value);
};

export const isMessageSuitedForFilters = (m, filters) => {
	return isEmptyOrContains(filters.packageNames, m.shortGroupName) &&
		isEmptyOrContains(filters.hostNames, m.host) &&
		isEmptyOrContains(filters.threadNames, m.thread);
};

export const filterMessages = (messages, filters) => {
	return messages.filter(m => isMessageSuitedForFilters(m, filters));
};

export const emptyFilters = () => {
	return buildFilters(Set(), Set(), Set());
};

export const buildFilters = (hostNames, packageNames, threadNames) => {
	return { hostNames, packageNames, threadNames };
};

export const emptyConnectionParams = () => {
	return {
		key: '', value: '', prefix: '', severity: 'INFO'
	};
};

export const newConnectionParams = (key, value, prefix, severity) => {
	return { key, value, prefix, severity };
};

export const updateFiltersValues = (current, filters, message) => {

	const updateFilterValues = (name, value) => {
		let res = current[name];
		if (isMessageSuitedForFilters(message, { ...filters, [name]: new Set() })) {
			res = res.update(value, 0, v => v + 1);
		}
		res = res.has(value) ? res : res.set(value, 0);
		return res;
	};
	return {
		hostNames: updateFilterValues('hostNames', message.host),
		packageNames: updateFilterValues('packageNames', message.shortGroupName),
		threadNames: updateFilterValues('threadNames', message.thread)
	};
};

export const createFiltersValues = (all, filters) => {
	return all.reduce((filtersValues, m) => updateFiltersValues(filtersValues, filters, m), emptyFilterValues());
};

export const emptyFilterValues = () => {
	return buildFilterValues(Map(), Map(), Map());
};

export const buildFilterValues = (hostNames, packageNames, threadNames) => {
	return { hostNames, packageNames, threadNames };
};

export const emptyMessages = () => {
	return {
		all: new List(),
		filtered: new List(),
		filters: emptyFilters(),
		filtersValues: emptyFilterValues(),
		nextMessageId: 1
	}
};