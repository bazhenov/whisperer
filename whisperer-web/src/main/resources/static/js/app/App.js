/**
 * Created with JetBrains PhpStorm.
 * User: antonhola
 * Date: 22/12/15
 * Time: 15:01
 * To change this template use File | Settings | File Templates.
 */

define(
	['underscore', 'backbone', 'FeedReceiver', 'FeedView', 'TraceCriteriaForm'],
	function (_, Backbone, FeedReceiver, FeedView, TraceCriteriaForm) {
		/**
		 *
		 * @constructor
		 */
		function App() {
			this.form = new TraceCriteriaForm();
			this.feedReceiver = new FeedReceiver();
			this.feedView = new FeedView();
		}

		_.extend(App.prototype, Backbone.Events, {
			run: function () {
				this.listenTo(this.feedReceiver, 'log-received', this.appendFeedRecord_);
				this.listenTo(this.form, 'start-listen', this.startReceiver_);
				this.listenTo(this.form, 'stop-listen', this.stopReceiver_);
			},

			appendFeedRecord_: function (record) {
				this.feedView.appendRecord(record);
				componentHandler.upgradeAllRegistered();
			},

			startReceiver_: function (criteria) {
				this.feedReceiver.start(criteria);
			},

			stopReceiver_: function () {
				this.feedReceiver.stop();
			}
		});

		return App;
	}
);