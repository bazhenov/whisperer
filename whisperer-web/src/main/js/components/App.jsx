import React, {Component} from 'react';
import Form from './Form.jsx'
import Messages from './Messages.jsx'
import Filters from './Filters.jsx';
import MessagesCount from './MessagesCount.jsx';
import Controls from './Controls.jsx'
import ToManyMessagesAlert from './ToManyMessagesAlert.jsx'
import { connect } from "react-redux";
import {
	startListening,
	stopListening,
	updateConnectionParams,
	loadConnectionParams,
	connectToSSE,
	disconnectFromSSE,
	clearMessages,
	updateFilters
} from '../actions';
import { MAX_MESSAGES_TO_DISPLAY } from '../constants';

export class App extends Component {

	constructor(props) {
		super(props);
		this.state = App._stateFromProps(props);
		this.updateCurrentConnectionParams = this.updateCurrentConnectionParams.bind(this);
		this.handleStartListening = this.handleStartListening.bind(this);
	}

	static _stateFromProps(props) {
		return {
			currentConnectionParams: { ...props.connectionParams }
		};
	}

	updateCurrentConnectionParams(key, value) {
		const newState = { ...this.state };
		newState.currentConnectionParams[key] = value;
		this.setState(newState);
	}

	componentWillReceiveProps(props) {
		const newConnectionParams = App._stateFromProps(props);
		if (!(Object.is(this.state.currentConnectionParams, newConnectionParams))) {
			this.setState(newConnectionParams);
		}
	}

	componentDidMount() {
		const { loadConnectionParams } = this.props;
		loadConnectionParams();
	}

	handleStartListening() {
		this.props.startListening(this.state.currentConnectionParams);
	}

	render() {
		const {
			stopListening,
			clearMessages,
			isListening,
			messages,
			updateFilters
		} = this.props;
		const { all, filtered, filters, filtersValues } = messages;
		const totalMessages = all.size;
		const filteredCount = filtered.size;

		const messagesToDisplay = filtered.slice(0, MAX_MESSAGES_TO_DISPLAY);

		return <div>
			<nav className="navbar navbar-default">
				<div className="container-fluid">
					<div className="navbar-header">
						<a className="navbar-brand" href="/">Whisperer</a>
					</div>
				</div>
			</nav>

			<div className="row">
				<div className="col-md-4">
					<Form updateCurrentConnectionParams={this.updateCurrentConnectionParams} isListening={isListening}
								connectionParams={this.state.currentConnectionParams} />
					<Controls isListening={isListening} stopListening={stopListening} clearMessages={clearMessages}
										startListening={this.handleStartListening} hasMessages={totalMessages > 0} />
				</div>
				<div className="col-md-8">
					<Filters filters={filters} updateFilters={updateFilters} filtersValues={filtersValues} />
				</div>
			</div>

			<div className="row">
				<div className="col-md-12">
					<MessagesCount totalCount={totalMessages} filteredCount={filteredCount} />
					<Messages messages={messagesToDisplay} />
				</div>
			</div>
			<ToManyMessagesAlert filteredCount={filteredCount} />
		</div>;
	}
}

const mapStateToProps = (state) => {
	return {...state};
};

const mapDispatchToProps = (dispatch) => {
	return {
		startListening: (connectionParams) => {
			dispatch(
				updateConnectionParams(connectionParams)
			);
			dispatch(startListening());
			dispatch(connectToSSE())
		},
		stopListening: () => {
			dispatch(stopListening());
			dispatch(disconnectFromSSE());
		},
		loadConnectionParams: () => {
			dispatch(loadConnectionParams())
		},
		clearMessages: () => {
			dispatch(clearMessages());
		},
		updateFilters: (filters) => {
			dispatch(updateFilters(filters));
		}
	}
};

export default connect(mapStateToProps, mapDispatchToProps)(App);