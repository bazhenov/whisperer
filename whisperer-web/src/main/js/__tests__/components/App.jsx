import React from "react";
import { App } from "../../components/App.jsx";
import renderer from "react-test-renderer";
import { connectionParams, message, message2 } from '../fixtures';
import { buildFilters, buildFilterValues } from '../../utils'
import { Set, Map } from 'immutable'
import * as constants from '../../constants'

describe('App', () => {
	const originalMaxMessagesCount = constants.MAX_MESSAGES_COUNT;

	const props = {
		isListening: true,
		connectionParams: connectionParams,
		messages: {
			all: [message, message2],
			filtered: [message],
			filters: buildFilters(
				Set(['http://search-coord5.akod.loc', 'http://search-coord4.akod.loc']),
				Set(['c.f.s.CoordinatorSearchEngine']),
				Set(['qtp-12345', 'qtp-23456']),
			),
			filtersValues: buildFilterValues(
				Map({'http://search-coord5.akod.loc': 5, 'http://search-coord4.akod.loc': 4, 'http://search-node68.akod.loc': 3}),
				Map({'c.f.s.CoordinatorSearchEngine': 5, 'c.f.s.i.IndexNodeImpl': 2}),
				Map({'qtp-12345':6, 'qtp-23456':5, 'qtp1914572623-35': 4, 'another-th': 0})
			)
		},
		loadConnectionParams: () => {}
	};

	it('regular', () => {
		expect(renderer.create(
			<App { ...props } />
		).toJSON()).toMatchSnapshot();
	});
	it('to many message to display', () => {
		constants.MAX_MESSAGES_COUNT = 1;
		expect(renderer.create(
			<App { ...props } filtered={[message, message2]} />
		).toJSON()).toMatchSnapshot();
	});
	afterEach(() => {
		constants.MAX_MESSAGES_COUNT = originalMaxMessagesCount;
	});
});