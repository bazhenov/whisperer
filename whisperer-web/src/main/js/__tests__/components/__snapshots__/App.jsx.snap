// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App regular 1`] = `
<div>
  <nav
    className="navbar navbar-default"
  >
    <div
      className="container-fluid"
    >
      <div
        className="navbar-header"
      >
        <a
          className="navbar-brand"
          href="/"
        >
          Whisperer
        </a>
      </div>
    </div>
  </nav>
  <div
    className="row"
  >
    <div
      className="col-md-4"
    >
      <div>
        <h3>
          Connection parameters
        </h3>
        <form
          onSubmit={[Function]}
        >
          <fieldset
            disabled={true}
          >
            <div
              className="form-group"
            >
              N
              <label
                htmlFor="mdcKeyInput"
              >
                MDC key name
              </label>
              <input
                className="form-control"
                id="mdcKeyInput"
                name="key"
                onChange={[Function]}
                type="text"
                value="clientId"
              />
            </div>
            <div
              className="form-group"
            >
              <label
                htmlFor="mdcValueInput"
              >
                Expected value
              </label>
              <input
                className="form-control"
                id="mdcValueInput"
                name="value"
                onChange={[Function]}
                type="text"
                value="some_ring"
              />
            </div>
            <div
              className="form-group"
            >
              <label
                htmlFor="logPrefixInput"
              >
                Log prefix
              </label>
              <input
                className="form-control"
                id="logPrefixInput"
                name="prefix"
                onChange={[Function]}
                type="text"
                value="com.farpost.search"
              />
            </div>
          </fieldset>
          <div
            className="form-group"
          >
            <div>
              <label>
                Severity
              </label>
            </div>
            <span
              className="choosed-severity label label-primary"
            >
              INFO
            </span>
            <span
              className="severity-button"
            >
              WARN
            </span>
            <span
              className="severity-button"
            >
              ERROR
            </span>
            <span
              className="severity-button"
            >
              DEBUG
            </span>
            <span
              className="severity-button"
            >
              TRACE
            </span>
          </div>
        </form>
      </div>
      <div
        id="control-buttons"
      >
        <a
          className="btn btn-default"
          href="#"
          onClick={[Function]}
        >
          STOP
        </a>
        
      </div>
    </div>
    <div
      className="col-md-8"
    >
      <div>
        <h3>
          Filters
        </h3>
        <div
          className="row"
        >
          <div
            className="col-xs-4"
          >
            <div
              className="form-group"
            >
              <label
                htmlFor="packageNameFilter"
              >
                Package Name
              </label>
              <input
                className="form-control"
                id="packageNameFilter"
                name="packageNameFilter"
                onChange={[Function]}
                type="text"
                value=""
              />
            </div>
            <div
              className="filters-filter-options"
            >
              <div
                className="checkbox filters-checkbox"
              >
                <label
                  className=""
                >
                  <input
                    checked={true}
                    name="packageNames"
                    onChange={[Function]}
                    type="checkbox"
                    value="c.f.s.CoordinatorSearchEngine"
                  />
                  c.f.s.CoordinatorSearchEngine
                </label>
                <div
                  className="filters-count"
                >
                  <span
                    className="label label-default"
                  >
                    5
                  </span>
                </div>
              </div>
              <div
                className="checkbox filters-checkbox"
              >
                <label
                  className=""
                >
                  <input
                    checked={false}
                    name="packageNames"
                    onChange={[Function]}
                    type="checkbox"
                    value="c.f.s.i.IndexNodeImpl"
                  />
                  c.f.s.i.IndexNodeImpl
                </label>
                <div
                  className="filters-count"
                >
                  <span
                    className="label label-default"
                  >
                    2
                  </span>
                </div>
              </div>
            </div>
            <div
              className="filter-labels"
            >
              <div>
                <span
                  className="filter-label-to-remove label label-info"
                >
                  <a
                    className="close remove-filter"
                    onClick={[Function]}
                    type="button"
                  >
                    <span
                      aria-hidden="true"
                    >
                      ×
                    </span>
                  </a>
                  c.f.s.CoordinatorSearchEngine
                </span>
              </div>
            </div>
          </div>
          <div
            className="col-xs-4"
          >
            <div
              className="form-group"
            >
              <label
                htmlFor="hostNameFilter"
              >
                Host Name
              </label>
              <input
                className="form-control"
                id="hostNameFilter"
                name="hostNameFilter"
                onChange={[Function]}
                type="text"
                value=""
              />
            </div>
            <div
              className="filters-filter-options"
            >
              <div
                className="checkbox filters-checkbox"
              >
                <label
                  className=""
                >
                  <input
                    checked={true}
                    name="hostNames"
                    onChange={[Function]}
                    type="checkbox"
                    value="http://search-coord4.akod.loc"
                  />
                  http://search-coord4.akod.loc
                </label>
                <div
                  className="filters-count"
                >
                  <span
                    className="label label-default"
                  >
                    4
                  </span>
                </div>
              </div>
              <div
                className="checkbox filters-checkbox"
              >
                <label
                  className=""
                >
                  <input
                    checked={true}
                    name="hostNames"
                    onChange={[Function]}
                    type="checkbox"
                    value="http://search-coord5.akod.loc"
                  />
                  http://search-coord5.akod.loc
                </label>
                <div
                  className="filters-count"
                >
                  <span
                    className="label label-default"
                  >
                    5
                  </span>
                </div>
              </div>
              <div
                className="checkbox filters-checkbox"
              >
                <label
                  className=""
                >
                  <input
                    checked={false}
                    name="hostNames"
                    onChange={[Function]}
                    type="checkbox"
                    value="http://search-node68.akod.loc"
                  />
                  http://search-node68.akod.loc
                </label>
                <div
                  className="filters-count"
                >
                  <span
                    className="label label-default"
                  >
                    3
                  </span>
                </div>
              </div>
            </div>
            <div
              className="filter-labels"
            >
              <div>
                <span
                  className="filter-label-to-remove label label-info"
                >
                  <a
                    className="close remove-filter"
                    onClick={[Function]}
                    type="button"
                  >
                    <span
                      aria-hidden="true"
                    >
                      ×
                    </span>
                  </a>
                  http://search-coord5.akod.loc
                </span>
              </div>
              <div>
                <span
                  className="filter-label-to-remove label label-info"
                >
                  <a
                    className="close remove-filter"
                    onClick={[Function]}
                    type="button"
                  >
                    <span
                      aria-hidden="true"
                    >
                      ×
                    </span>
                  </a>
                  http://search-coord4.akod.loc
                </span>
              </div>
            </div>
          </div>
          <div
            className="col-xs-4"
          >
            <div
              className="form-group"
            >
              <label
                htmlFor="threadNameFilter"
              >
                Thread Name
              </label>
              <input
                className="form-control"
                id="threadNameFilter"
                name="threadNameFilter"
                onChange={[Function]}
                type="text"
                value=""
              />
            </div>
            <div
              className="filters-filter-options"
            >
              <div
                className="checkbox filters-checkbox"
              >
                <label
                  className=""
                >
                  <input
                    checked={true}
                    name="threadNames"
                    onChange={[Function]}
                    type="checkbox"
                    value="qtp-12345"
                  />
                  qtp-12345
                </label>
                <div
                  className="filters-count"
                >
                  <span
                    className="label label-default"
                  >
                    6
                  </span>
                </div>
              </div>
              <div
                className="checkbox filters-checkbox"
              >
                <label
                  className=""
                >
                  <input
                    checked={true}
                    name="threadNames"
                    onChange={[Function]}
                    type="checkbox"
                    value="qtp-23456"
                  />
                  qtp-23456
                </label>
                <div
                  className="filters-count"
                >
                  <span
                    className="label label-default"
                  >
                    5
                  </span>
                </div>
              </div>
              <div
                className="checkbox filters-checkbox"
              >
                <label
                  className=""
                >
                  <input
                    checked={false}
                    name="threadNames"
                    onChange={[Function]}
                    type="checkbox"
                    value="qtp1914572623-35"
                  />
                  qtp1914572623-35
                </label>
                <div
                  className="filters-count"
                >
                  <span
                    className="label label-default"
                  >
                    4
                  </span>
                </div>
              </div>
              <div
                className="checkbox filters-checkbox"
              >
                <label
                  className="zero-count-filter"
                >
                  <input
                    checked={false}
                    name="threadNames"
                    onChange={[Function]}
                    type="checkbox"
                    value="another-th"
                  />
                  another-th
                </label>
                <div
                  className="filters-count"
                >
                  <span
                    className="label label-default"
                  >
                    0
                  </span>
                </div>
              </div>
            </div>
            <div
              className="filter-labels"
            >
              <div>
                <span
                  className="filter-label-to-remove label label-info"
                >
                  <a
                    className="close remove-filter"
                    onClick={[Function]}
                    type="button"
                  >
                    <span
                      aria-hidden="true"
                    >
                      ×
                    </span>
                  </a>
                  qtp-12345
                </span>
              </div>
              <div>
                <span
                  className="filter-label-to-remove label label-info"
                >
                  <a
                    className="close remove-filter"
                    onClick={[Function]}
                    type="button"
                  >
                    <span
                      aria-hidden="true"
                    >
                      ×
                    </span>
                  </a>
                  qtp-23456
                </span>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="row"
  >
    <div
      className="col-md-12"
    >
      <div
        id="messages-records-count"
      >
        <strong />
        
         records
      </div>
      <div
        className="row"
      >
        <div
          className="col-md-12"
        >
          <div
            className="row message-row"
          >
            <div
              className="col-xs-3 message-row-left-col"
            >
              <div>
                <small>
                  14:17:09.683
                </small>
              </div>
              <div
                className="message-host-name"
              >
                <strong>
                  http://search-coord5.akod.loc
                </strong>
              </div>
              <div>
                <div
                  className="message-tooltip"
                >
                  <div
                    className="message-group-name"
                  >
                    c.f.s.CoordinatorSearchEngine
                  </div>
                  <span
                    className="message-tooltip-text"
                  >
                    com.farpost.search.CoordinatorSearchEngine
                  </span>
                </div>
              </div>
              <div
                className="message-thread-name"
              >
                thread: 
                qtp-12345
              </div>
            </div>
            <div
              className="col-xs-9 message-row-right-col"
            >
              <div
                className="message-block"
              >
                <span>
                  Delegating execution to command 
                </span>
                <a
                  href="#"
                  onClick={[Function]}
                >
                  {
                  ...
                  }
                </a>
                <span>
                  , query: '
                </span>
                <a
                  href="#"
                  onClick={[Function]}
                >
                  {
                  ...
                  }
                </a>
                <span>
                  ', sort: 
                </span>
                <span>
                  null
                </span>
                <span>
                  , facets: 
                </span>
                <a
                  href="#"
                  onClick={[Function]}
                >
                  {
                  ...
                  }
                </a>
                <span>
                  , filter: '
                </span>
                <a
                  href="#"
                  onClick={[Function]}
                >
                  {
                  ...
                  }
                </a>
                <span>
                  '
                </span>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="row"
  >
    <div
      className="col-md-12"
    >
      <div
        className="alert alert-warning"
        role="alert"
      >
        Too many messages! Only first 
        1000
         was showed.
      </div>
    </div>
  </div>
</div>
`;

exports[`App to many message to display 1`] = `
<div>
  <nav
    className="navbar navbar-default"
  >
    <div
      className="container-fluid"
    >
      <div
        className="navbar-header"
      >
        <a
          className="navbar-brand"
          href="/"
        >
          Whisperer
        </a>
      </div>
    </div>
  </nav>
  <div
    className="row"
  >
    <div
      className="col-md-4"
    >
      <div>
        <h3>
          Connection parameters
        </h3>
        <form
          onSubmit={[Function]}
        >
          <fieldset
            disabled={true}
          >
            <div
              className="form-group"
            >
              N
              <label
                htmlFor="mdcKeyInput"
              >
                MDC key name
              </label>
              <input
                className="form-control"
                id="mdcKeyInput"
                name="key"
                onChange={[Function]}
                type="text"
                value="clientId"
              />
            </div>
            <div
              className="form-group"
            >
              <label
                htmlFor="mdcValueInput"
              >
                Expected value
              </label>
              <input
                className="form-control"
                id="mdcValueInput"
                name="value"
                onChange={[Function]}
                type="text"
                value="some_ring"
              />
            </div>
            <div
              className="form-group"
            >
              <label
                htmlFor="logPrefixInput"
              >
                Log prefix
              </label>
              <input
                className="form-control"
                id="logPrefixInput"
                name="prefix"
                onChange={[Function]}
                type="text"
                value="com.farpost.search"
              />
            </div>
          </fieldset>
          <div
            className="form-group"
          >
            <div>
              <label>
                Severity
              </label>
            </div>
            <span
              className="choosed-severity label label-primary"
            >
              INFO
            </span>
            <span
              className="severity-button"
            >
              WARN
            </span>
            <span
              className="severity-button"
            >
              ERROR
            </span>
            <span
              className="severity-button"
            >
              DEBUG
            </span>
            <span
              className="severity-button"
            >
              TRACE
            </span>
          </div>
        </form>
      </div>
      <div
        id="control-buttons"
      >
        <a
          className="btn btn-default"
          href="#"
          onClick={[Function]}
        >
          STOP
        </a>
        
      </div>
    </div>
    <div
      className="col-md-8"
    >
      <div>
        <h3>
          Filters
        </h3>
        <div
          className="row"
        >
          <div
            className="col-xs-4"
          >
            <div
              className="form-group"
            >
              <label
                htmlFor="packageNameFilter"
              >
                Package Name
              </label>
              <input
                className="form-control"
                id="packageNameFilter"
                name="packageNameFilter"
                onChange={[Function]}
                type="text"
                value=""
              />
            </div>
            <div
              className="filters-filter-options"
            >
              <div
                className="checkbox filters-checkbox"
              >
                <label
                  className=""
                >
                  <input
                    checked={true}
                    name="packageNames"
                    onChange={[Function]}
                    type="checkbox"
                    value="c.f.s.CoordinatorSearchEngine"
                  />
                  c.f.s.CoordinatorSearchEngine
                </label>
                <div
                  className="filters-count"
                >
                  <span
                    className="label label-default"
                  >
                    5
                  </span>
                </div>
              </div>
              <div
                className="checkbox filters-checkbox"
              >
                <label
                  className=""
                >
                  <input
                    checked={false}
                    name="packageNames"
                    onChange={[Function]}
                    type="checkbox"
                    value="c.f.s.i.IndexNodeImpl"
                  />
                  c.f.s.i.IndexNodeImpl
                </label>
                <div
                  className="filters-count"
                >
                  <span
                    className="label label-default"
                  >
                    2
                  </span>
                </div>
              </div>
            </div>
            <div
              className="filter-labels"
            >
              <div>
                <span
                  className="filter-label-to-remove label label-info"
                >
                  <a
                    className="close remove-filter"
                    onClick={[Function]}
                    type="button"
                  >
                    <span
                      aria-hidden="true"
                    >
                      ×
                    </span>
                  </a>
                  c.f.s.CoordinatorSearchEngine
                </span>
              </div>
            </div>
          </div>
          <div
            className="col-xs-4"
          >
            <div
              className="form-group"
            >
              <label
                htmlFor="hostNameFilter"
              >
                Host Name
              </label>
              <input
                className="form-control"
                id="hostNameFilter"
                name="hostNameFilter"
                onChange={[Function]}
                type="text"
                value=""
              />
            </div>
            <div
              className="filters-filter-options"
            >
              <div
                className="checkbox filters-checkbox"
              >
                <label
                  className=""
                >
                  <input
                    checked={true}
                    name="hostNames"
                    onChange={[Function]}
                    type="checkbox"
                    value="http://search-coord4.akod.loc"
                  />
                  http://search-coord4.akod.loc
                </label>
                <div
                  className="filters-count"
                >
                  <span
                    className="label label-default"
                  >
                    4
                  </span>
                </div>
              </div>
              <div
                className="checkbox filters-checkbox"
              >
                <label
                  className=""
                >
                  <input
                    checked={true}
                    name="hostNames"
                    onChange={[Function]}
                    type="checkbox"
                    value="http://search-coord5.akod.loc"
                  />
                  http://search-coord5.akod.loc
                </label>
                <div
                  className="filters-count"
                >
                  <span
                    className="label label-default"
                  >
                    5
                  </span>
                </div>
              </div>
              <div
                className="checkbox filters-checkbox"
              >
                <label
                  className=""
                >
                  <input
                    checked={false}
                    name="hostNames"
                    onChange={[Function]}
                    type="checkbox"
                    value="http://search-node68.akod.loc"
                  />
                  http://search-node68.akod.loc
                </label>
                <div
                  className="filters-count"
                >
                  <span
                    className="label label-default"
                  >
                    3
                  </span>
                </div>
              </div>
            </div>
            <div
              className="filter-labels"
            >
              <div>
                <span
                  className="filter-label-to-remove label label-info"
                >
                  <a
                    className="close remove-filter"
                    onClick={[Function]}
                    type="button"
                  >
                    <span
                      aria-hidden="true"
                    >
                      ×
                    </span>
                  </a>
                  http://search-coord5.akod.loc
                </span>
              </div>
              <div>
                <span
                  className="filter-label-to-remove label label-info"
                >
                  <a
                    className="close remove-filter"
                    onClick={[Function]}
                    type="button"
                  >
                    <span
                      aria-hidden="true"
                    >
                      ×
                    </span>
                  </a>
                  http://search-coord4.akod.loc
                </span>
              </div>
            </div>
          </div>
          <div
            className="col-xs-4"
          >
            <div
              className="form-group"
            >
              <label
                htmlFor="threadNameFilter"
              >
                Thread Name
              </label>
              <input
                className="form-control"
                id="threadNameFilter"
                name="threadNameFilter"
                onChange={[Function]}
                type="text"
                value=""
              />
            </div>
            <div
              className="filters-filter-options"
            >
              <div
                className="checkbox filters-checkbox"
              >
                <label
                  className=""
                >
                  <input
                    checked={true}
                    name="threadNames"
                    onChange={[Function]}
                    type="checkbox"
                    value="qtp-12345"
                  />
                  qtp-12345
                </label>
                <div
                  className="filters-count"
                >
                  <span
                    className="label label-default"
                  >
                    6
                  </span>
                </div>
              </div>
              <div
                className="checkbox filters-checkbox"
              >
                <label
                  className=""
                >
                  <input
                    checked={true}
                    name="threadNames"
                    onChange={[Function]}
                    type="checkbox"
                    value="qtp-23456"
                  />
                  qtp-23456
                </label>
                <div
                  className="filters-count"
                >
                  <span
                    className="label label-default"
                  >
                    5
                  </span>
                </div>
              </div>
              <div
                className="checkbox filters-checkbox"
              >
                <label
                  className=""
                >
                  <input
                    checked={false}
                    name="threadNames"
                    onChange={[Function]}
                    type="checkbox"
                    value="qtp1914572623-35"
                  />
                  qtp1914572623-35
                </label>
                <div
                  className="filters-count"
                >
                  <span
                    className="label label-default"
                  >
                    4
                  </span>
                </div>
              </div>
              <div
                className="checkbox filters-checkbox"
              >
                <label
                  className="zero-count-filter"
                >
                  <input
                    checked={false}
                    name="threadNames"
                    onChange={[Function]}
                    type="checkbox"
                    value="another-th"
                  />
                  another-th
                </label>
                <div
                  className="filters-count"
                >
                  <span
                    className="label label-default"
                  >
                    0
                  </span>
                </div>
              </div>
            </div>
            <div
              className="filter-labels"
            >
              <div>
                <span
                  className="filter-label-to-remove label label-info"
                >
                  <a
                    className="close remove-filter"
                    onClick={[Function]}
                    type="button"
                  >
                    <span
                      aria-hidden="true"
                    >
                      ×
                    </span>
                  </a>
                  qtp-12345
                </span>
              </div>
              <div>
                <span
                  className="filter-label-to-remove label label-info"
                >
                  <a
                    className="close remove-filter"
                    onClick={[Function]}
                    type="button"
                  >
                    <span
                      aria-hidden="true"
                    >
                      ×
                    </span>
                  </a>
                  qtp-23456
                </span>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="row"
  >
    <div
      className="col-md-12"
    >
      <div
        id="messages-records-count"
      >
        <strong />
        
         records
      </div>
      <div
        className="row"
      >
        <div
          className="col-md-12"
        >
          <div
            className="row message-row"
          >
            <div
              className="col-xs-3 message-row-left-col"
            >
              <div>
                <small>
                  14:17:09.683
                </small>
              </div>
              <div
                className="message-host-name"
              >
                <strong>
                  http://search-coord5.akod.loc
                </strong>
              </div>
              <div>
                <div
                  className="message-tooltip"
                >
                  <div
                    className="message-group-name"
                  >
                    c.f.s.CoordinatorSearchEngine
                  </div>
                  <span
                    className="message-tooltip-text"
                  >
                    com.farpost.search.CoordinatorSearchEngine
                  </span>
                </div>
              </div>
              <div
                className="message-thread-name"
              >
                thread: 
                qtp-12345
              </div>
            </div>
            <div
              className="col-xs-9 message-row-right-col"
            >
              <div
                className="message-block"
              >
                <span>
                  Delegating execution to command 
                </span>
                <a
                  href="#"
                  onClick={[Function]}
                >
                  {
                  ...
                  }
                </a>
                <span>
                  , query: '
                </span>
                <a
                  href="#"
                  onClick={[Function]}
                >
                  {
                  ...
                  }
                </a>
                <span>
                  ', sort: 
                </span>
                <span>
                  null
                </span>
                <span>
                  , facets: 
                </span>
                <a
                  href="#"
                  onClick={[Function]}
                >
                  {
                  ...
                  }
                </a>
                <span>
                  , filter: '
                </span>
                <a
                  href="#"
                  onClick={[Function]}
                >
                  {
                  ...
                  }
                </a>
                <span>
                  '
                </span>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="row"
  >
    <div
      className="col-md-12"
    >
      <div
        className="alert alert-warning"
        role="alert"
      >
        Too many messages! Only first 
        1000
         was showed.
      </div>
    </div>
  </div>
</div>
`;
